assets/js/googlesitekit/data/create-settings-store.test.js

```diff
@@ -0,0 +1,453 @@
+
+				dispatch.setSettings( clientValues );
+				dispatch.receiveSettings( serverValues );
+
+				// Client values take precedence if they were already modified before receiving from the server.
+				expect( store.getState().settings ).toMatchObject( { ...serverValues, ...clientValues } );
+			} );
+		} );
+
+		describe( 'saveSettings', () => {
+			it( 'does not require any params', () => {
+				expect( async () => {
+					fetch
+						.doMockOnceIf(
+							/^\/google-site-kit\/v1\/core\/site\/data\/settings/
+						)
+						.mockResponseOnce(
+							JSON.stringify( { setting1: 'serverside' } ),
+							{ status: 200 }
+						);
+
+					await dispatch.saveSettings();
+				} ).not.toThrow();
+			} );
+
+			it( 'updates settings from server', async () => {
+				const response = { setting1: 'serverside' };
+				fetch
+					.doMockIf(
+						/^\/google-site-kit\/v1\/core\/site\/data\/settings/
+					)
+					.mockResponse(
+						JSON.stringify( response ),
+						{ status: 200 }
+					);
+
+				// The server is the authority. So because this won't be part of the response
+				// (see above), it will be disregarded.
+				dispatch.setSettings( { setting2: 'clientside' } );
+
+				await dispatch.saveSettings();
+				// Two fetch requests, one to get settings, the other to update.
+				expect( fetch ).toHaveBeenCalledTimes( 2 );
+
+				expect( store.getState().settings ).toMatchObject( response );
+			} );
+		} );
+
+		describe( 'fetchSaveSettings', () => {
+			it( 'requires the values param', () => {
+				expect( () => {
+					dispatch.receiveSaveSettings();
+				} ).toThrow( 'values is required.' );
+			} );
+
+			it( 'sets isDoingSaveSettings', () => {
+				fetch
+					.doMockOnceIf(
+						/^\/google-site-kit\/v1\/core\/site\/data\/settings/
+					)
+					.mockResponseOnce(
+						JSON.stringify( { setting1: true } ),
+						{ status: 200 }
+					);
+
+				dispatch.fetchSaveSettings( {} );
+				expect( select.isDoingSaveSettings() ).toEqual( true );
+			} );
+		} );
+
+		describe( 'receiveSaveSettings', () => {
+			it( 'requires the values param', () => {
+				expect( () => {
+					dispatch.receiveSaveSettings();
+				} ).toThrow( 'values is required.' );
+			} );
+
+			it( 'receives and sets values', () => {
+				const serverValues = { setting1: 'serverside', setting2: 'serverside' };
+				const clientValues = { setting1: 'clientside', setting3: 'clientside' };
+
+				dispatch.setSettings( clientValues );
+				dispatch.receiveSaveSettings( serverValues );
+
+				// Client values are ignored here, server values replace them.
+				expect( store.getState().settings ).toMatchObject( { ...serverValues } );
+			} );
+		} );
+
+		// Tests for "pseudo-action" setSetting, available via setting-specific "set{SettingSlug}".
+		describe( 'setSetting', () => {
+			it( 'has the correct action name', () => {
+				expect( Object.keys( storeDefinition.actions ) ).toEqual(
+					expect.arrayContaining( [ SETTING.action ] )
```

Comment:
I know this variable was defined above, but I think this abstraction makes it a lot harder to understand precisely what this is testing. Consider:

```suggestion
					expect.arrayContaining( [ 'setTestSetting' ] )
```

I know it is "redundant" in the strictest sense, but the test as-is compares two variables not in-view in my VSCode window on a 5K display, so I find it to lack context compared to my suggestion.

Reply:
(The comment helps a bit, but I think we can be clearer 😄)